src/test/java/org/cactoos/func/CallableOfTest.java

```diff
@@ -47,4 +50,39 @@ public void convertsFuncIntoCallable() throws Exception {
         );
     }
 
+    @Test
+    public void convertsRunnableIntoCallable() throws Exception {
+        final AtomicBoolean flag = new AtomicBoolean(false);
+        new CallableOf<>(
+            (Runnable) () -> { flag.set(true); }
+        ).call();
+        MatcherAssert.assertThat(
+            flag.get(),
+            Matchers.is(true)
+        );
+    }
+
+    @Test
+    public void convertsProcIntoCallable() throws Exception {
+        final AtomicBoolean flag = new AtomicBoolean(false);
+        new CallableOf<>(
+            (Proc<AtomicBoolean>) (unused) -> { flag.set(true); }
```

Comment:
@llorllale What is this ``unused``? What is its type and how could we make sure that ``CallableOf`` doesn't change it before passing it to the Proc's method?

Reply:
@amihaiemil I'm actually not sure what the purpose of having a ctor accepting a `Proc` is.

`Proc` is a functional interface with its SAM like this:
```java
void exec(X input) throws Exception;
```

`CallableOf` is a `Callable` and thus its `call()` takes no arguments. I can't think of a decent way to make this test make any more sense.

Reply:
@llorllale hmm, yes, I see ``CallableOf`` wraps it inside FuncOf:  ``this(new FuncOf<>(proc));`` -- then we should see what FuncOf does with the Proc, if it uses that parameter somehow. If not, then I think it should be taken out.

We can leave if like this and maybe you can open an issue and ask clarifications? (since you noticed this in the first place)